libraries/chain/controller.cpp

```diff
@@ -1879,7 +1879,9 @@ struct controller_impl {
          block_state_ptr bsp = block_state_future.get();
          const auto& b = bsp->block;
 
-         if( conf.terminate_at_block > 0 && conf.terminate_at_block < b->block_num() ) {
+         if( conf.terminate_at_block > 0 &&
+            ( (conf.read_mode != db_read_mode::IRREVERSIBLE && conf.terminate_at_block < b->block_num() ) ||
+              ( conf.read_mode == db_read_mode::IRREVERSIBLE && conf.terminate_at_block <= self.last_irreversible_block_num() ) )  ) {
```

Comment:
shouldn't this be '<', like the head/read only mode check, so that it terminates after applying the block, rather than before applying the block?

Reply:
After test, it is currently works well.
let's assume terminate_at_block is 50.
In case of conf.read_mode != db_read_mode::IRREVERSIBLE,  when b->block_num() is 51, stop, then the chain already have 50.  
In case of conf.read_mode == db_read_mode::IRREVERSIBLE, when self.last_irreversible_block_num()  is 50, then stop, this time the chain already has LIB 50, and the head is a bigger number we don't care. 

Where's wrong you think?